core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -148,7 +148,11 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** Cleanup local dirs and stop shuffle sender. */
   private[spark] def stop() {
     // Remove the shutdown hook.  It causes memory leaks if we leave it around.
-    Utils.removeShutdownHook(shutdownHook)
+    try {
+      Utils.removeShutdownHook(shutdownHook)
```

Comment:
Hmm. It should be ok to allow removing the hooks during shutdown. `SparkShutdownHookManager.runAll` is safe against modifications to the underlying queue.